ctru-rs/src/services/soc.rs

```diff
@@ -52,14 +53,49 @@ impl Soc {
             },
         )?;
 
-        Ok(Self { _service_handler })
+        Ok(Self {
+            _service_handler,
+            sock_3dslink: -1,
+        })
     }
 
     /// IP Address of the Nintendo 3DS system.
     pub fn host_address(&self) -> Ipv4Addr {
         let raw_id = unsafe { libc::gethostid() };
         Ipv4Addr::from(raw_id.to_ne_bytes())
     }
+
+    /// Redirect output streams (i.e. [`println`] and [`eprintln`]) to the `3dslink` server.
+    /// Requires `3dslink` >= 0.6.1 and `new-hbmenu` >= 2.3.0.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if a connection cannot be established to the server, or
+    /// output was already previously redirected.
+    pub fn redirect_to_3dslink(&mut self, stdout: bool, stderr: bool) -> crate::Result<()> {
+        if self.sock_3dslink >= 0 {
+            // TODO AlreadyRedirected or something
+            return Err(crate::Error::ServiceAlreadyActive);
```

Comment:
oops, forgot to address this as well. Should this be a new enum variant for `crate::Error`, or maybe this function should not return `crate::Result` at all?

Reply:
Either re-redirect it (close the socket and reopen it?) Or return an error with a new variant. It'll probably depend on if libctru supports calling it twice.

Reply:
After some testing, it seems like either the client or the server implementation doesn't support more than one call. So I think I will just add a new enum variant to be safe.